### PR TITLE
Add force stop storage engine interface

### PR DESCRIPTION
**Pull request by qiaojialin**
The shut down is not simulated because syncCloseAllProcessors in StorageEngine.stop(), need to refactor

### CODE REVIEW

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBRecoverUnclosedIT.java [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        Assert.assertEquals(3, cnt);
+      }
+
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail(e.getMessage());
+    }
+
+    EnvironmentUtils.shutdownDaemon();
+    EnvironmentUtils.reactiveDaemon();
```
Comment (JackieTien97) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBRecoverUnclosedIT.java:155]:
```suggestion
    EnvironmentUtils.restartDaemon();
```

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBRecoverUnclosedIT.java [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    
+    insertMoreData();
+
+    EnvironmentUtils.shutdownDaemon();
+    EnvironmentUtils.reactiveDaemon();
```
Comment (JackieTien97) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBRecoverUnclosedIT.java:160]:
```suggestion
    EnvironmentUtils.restartDaemon();
```

File: server/src/main/java/org/apache/iotdb/db/service/RegisterManager.java [hunk 3]
```diff
@@ -64,4 +65,21 @@ public void deregisterAll() {
...
+   * stop all service and clear iService list.
+   */
+  public void shutdownAll() {
+    //we stop JMXServer at last
+    Collections.reverse(iServices);
+    for (IService service : iServices) {
+      try {
+        service.shutdown(10000);
+      } catch (Exception e) {
+        logger.error("Failed to stop {} because:", service.getID().getName(), e);
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/service/RegisterManager.java:79]: I think you should rethrow the exception, and in the tests, you should catch it, and add failed() clause.

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 4]
```diff
@@ -176,6 +176,7 @@ public synchronized void init() {
     try {
       tagLogFile = new TagLogFile(config.getSchemaDir(), MetadataConstant.TAG_LOG);
 
+      isRecovering = true;
```
[the remark this answers is not in the document]
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:179]: Actually, when close and restart the IoTDB using API, we don't get a new Instance of MManager. If I didn't set `isRecovering = true` here, the value of `isRecovering` would be null.

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBRecoverUnclosedIT.java (hunk 1 above)
Comment (HTHou) [reply server/src/test/java/org/apache/iotdb/db/integration/IoTDBRecoverUnclosedIT.java:155]: Fixed, thanks~

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBRecoverUnclosedIT.java (hunk 2 above)
Comment (HTHou) [reply server/src/test/java/org/apache/iotdb/db/integration/IoTDBRecoverUnclosedIT.java:160]: Fixed~

File: server/src/main/java/org/apache/iotdb/db/service/RegisterManager.java (hunk 3 above)
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/service/RegisterManager.java:79]: Very good point! Thanks.

**Pull request merged**